x-pack/plugins/apm/server/lib/service_map/get_service_map_from_trace_ids.test.ts

```diff
@@ -6,23 +6,180 @@
+        serviceName: 'opbeans-node',
+        environment: undefined,
+      });
+
+      const connectionsPairs = getConnectionsPairs(connections);
+
+      expect(connectionsPairs).toEqual([
+        'opbeans-node:production -> opbeans-python:production',
+        'opbeans-node:production -> opbeans-ruby:production',
+        'opbeans-python:null -> opbeans-node:null',
+        'opbeans-python:production -> opbeans-node:production',
+        'opbeans-python:production -> opbeans-ruby:production',
+        'opbeans-ruby:production -> opbeans-node:production',
+        'opbeans-ruby:production -> opbeans-python:production',
+      ]);
+    });
+  });
+
+  describe('if environment is given', () => {
+    it('shows all connections for testing environment', () => {
+      const response = serviceMapFromTraceIdsScriptResponse as PromiseReturnType<
+        typeof fetchServicePathsFromTraceIds
+      >;
+
+      const connections = getConnections({
+        paths: response.aggregations?.service_map.value.paths,
+        serviceName: undefined,
+        environment: 'testing',
+      });
+
+      const connectionsPairs = getConnectionsPairs(connections);
+
+      expect(connectionsPairs).toEqual([
+        'opbeans-go:testing -> opbeans-java:testing',
+      ]);
+    });
+    it('shows all connections for production environment', () => {
+      const response = serviceMapFromTraceIdsScriptResponse as PromiseReturnType<
+        typeof fetchServicePathsFromTraceIds
+      >;
+
+      const connections = getConnections({
+        paths: response.aggregations?.service_map.value.paths,
+        serviceName: undefined,
+        environment: 'production',
+      });
+
+      const connectionsPairs = getConnectionsPairs(connections);
+
+      expect(connectionsPairs).toEqual([
+        'opbeans-node:production -> opbeans-python:production',
+        'opbeans-node:production -> opbeans-ruby:production',
+        'opbeans-python:production -> opbeans-node:production',
+        'opbeans-python:production -> opbeans-ruby:production',
+        'opbeans-ruby:production -> opbeans-node:production',
+        'opbeans-ruby:production -> opbeans-python:production',
+      ]);
+    });
+  });
+
+  describe('if environment is "not defined"', () => {
+    it('shows all connections where environment is not set', () => {
+      const response = serviceMapFromTraceIdsScriptResponseEnvNotDefined as PromiseReturnType<
+        typeof fetchServicePathsFromTraceIds
+      >;
+      const connections = getConnections({
+        paths: response.aggregations?.service_map.value.paths,
+        serviceName: undefined,
+        environment: 'ENVIRONMENT_NOT_DEFINED',
+      });
+
+      const environments = new Set();
+
+      connections.forEach((conn) =>
+        environments.add(conn.source['service.environment'])
+      );
+      expect(
+        connections.some((conn) => conn.source['service.environment'] !== null)
+      ).toBeFalsy();
+    });
```

Comment:
Shouldn't this also use `getConnectionsPairs()`?

Reply:
yes, that will be better, I'll change the mocked data.